sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/NonStreamingOrderByUtils.java

```diff
@@ -0,0 +1,105 @@
+package com.azure.cosmos.implementation.query;
+
+import com.azure.cosmos.BridgeInternal;
+import com.azure.cosmos.implementation.ClientSideRequestStatistics;
+import com.azure.cosmos.implementation.Document;
+import com.azure.cosmos.implementation.ImplementationBridgeHelpers;
+import com.azure.cosmos.implementation.QueryMetrics;
+import com.azure.cosmos.implementation.RequestChargeTracker;
+import com.azure.cosmos.implementation.Resource;
+import com.azure.cosmos.implementation.query.orderbyquery.OrderByRowResult;
+import com.azure.cosmos.implementation.query.orderbyquery.OrderbyRowComparer;
+import com.azure.cosmos.models.ModelBridgeInternal;
+import reactor.core.publisher.Flux;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Function;
+
+public class NonStreamingOrderByUtils {
+    private final static
+    ImplementationBridgeHelpers.CosmosDiagnosticsHelper.CosmosDiagnosticsAccessor diagnosticsAccessor =
+        ImplementationBridgeHelpers.CosmosDiagnosticsHelper.getCosmosDiagnosticsAccessor();
+
+    public static <T extends Resource> Flux<OrderByRowResult<Document>> nonStreamingOrderedMerge(OrderbyRowComparer<Document> consumeComparer,
+                                                                                                 RequestChargeTracker tracker,
+                                                                                                 List<DocumentProducer<Document>> documentProducers,
+                                                                                                 int initialPageSize,
+                                                                                                 Map<String, QueryMetrics> queryMetricsMap,
+                                                                                                 Collection<ClientSideRequestStatistics> clientSideRequestStatistics) {
+        @SuppressWarnings("unchecked")
+        Flux<OrderByRowResult<Document>>[] fluxes = documentProducers
+            .subList(0, documentProducers.size())
+            .stream()
+            .map(producer ->
+                toNonStreamingOrderByQueryResultObservable(producer, tracker, queryMetricsMap, initialPageSize,
+                    consumeComparer, clientSideRequestStatistics))
+            .toArray(Flux[]::new);
+        return Flux.mergeOrdered(consumeComparer, fluxes);
```

Comment:
This is a deprecated API - please use the new one (if we really need this instead of using one PriorityQueue across all Document Producers)

Reply:
Also this is functionally incorrect - it does not sort - it only picks the smallest value from each source then the seocnd smallest value etc. - no ordering across sources.

Reply:
Please also try to get more info on how this would behave if number of sources is higher than SMALL_BUFFER (which seems to be the default prefetch count)

Reply:
We need to ensure that this isn't resulting in hangs like the groupBy operator

Reply:
As discussed offline using Flux.mergeComparingDelayError()

Just for reference used the below example. For now I am keeping the prefetch value to be 1. But I tested the code with different prefetch values, and still getting a sorted merged Flux as we want.

```Integer[] a1 = {2, 3, 6, 7, 8};
        Flux<Integer> flux1 = Flux.fromArray(a1);
        Integer[] a2 = {101, 103, 105, 107, 110};
        Flux<Integer> flux2 = Flux.fromArray(a2);
        Integer[] a3 = {1, 4, 5, 108, 115};
        Flux<Integer> flux3 = Flux.fromArray(a3);
        Flux<Integer>[] fluxArray = new Flux[] {flux1, flux2, flux3};

        Comparator<Integer> ascendingComparator = (a, b) -> a.compareTo(b);
        Flux<Integer> flux = Flux.mergeComparingDelayError(1, ascendingComparator, fluxArray);

        flux.subscribe(
            item -> System.out.println("Item: " + item),
            error -> System.err.println("Error: " + error),
            () -> System.out.println("Completed")
        );